cep-deprecation.md

```diff
@@ -0,0 +1,158 @@
+<table>
+<tr><td> Title </td><td> Conda Deprecation Schedule </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Ken Odegard &lt;kodegard@anaconda.com&gt; </td></tr>
+<tr><td> Created </td><td> May 20, 2022 </td></tr>
+<tr><td> Updated </td><td> July 20, 2022 </td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CEP describes a deprecation schedule to properly warn about upcoming removals from the codebase. This policy expands on ideas and terminology defined in the Conda Release Schedule (CEP-?).
+
+> **Note**
+> This CEP is only applicable for projects that have adopted the Conda Release Schedule (CEP-?).
+
+## Specification
+
+We propose a deprecation schedule that is slower than the Conda Release Schedule (CEP-?). This is in acknowledgment of our diverse user groups (e.g. everything from per user per machine installs to multi-user installs on shared clusters).
+
+We define a new type of release, a deprecation release, to augment regular releases. A deprecation release occurs three (3) times every year in January, May, and September:
+
+| Version | Release Type |
+|---|---|
+| `YY.1.0` | regular, **deprecation** |
+| `YY.2.0` | optional |
+| `YY.3.0` | regular |
+| `YY.4.0` | optional |
+| `YY.5.0` | regular, **deprecation** |
+| `YY.6.0` | optional |
+| `YY.7.0` | regular |
+| `YY.8.0` | optional |
+| `YY.9.0` | regular, **deprecation** |
+| `YY.10.0` | optional |
+| `YY.11.0` | regular |
+| `YY.12.0` | optional |
```

Comment:
I think 3 times per year makes the deprecation cycle too fast for most users that only update conda irregular, if at all.

I suggest to do a deprecation cycle twice every year until we have encouraged users to update conda more actively.

I would use March and September since both are far enough from various holidays and seasonal slowdowns away to make upgrades easier to handle.

The only downside IMO would be that the actual removal of deprecated features would move around on the calendar, given the three-step default.

```suggestion
We define a new type of release, a deprecation release, to augment regular releases. A deprecation release occurs twice every year in March and September:

| Version | Release Type |
|---|---|
| `YY.1.0` | regular |
| `YY.2.0` | optional |
| `YY.3.0` | regular, **deprecation** |
| `YY.4.0` | optional |
| `YY.5.0` | regular |
| `YY.6.0` | optional |
| `YY.7.0` | regular |
| `YY.8.0` | optional |
| `YY.9.0` | regular, **deprecation** |
| `YY.10.0` | optional |
| `YY.11.0` | regular |
| `YY.12.0` | optional |
```

What do you think?

Reply:
BTW pip's deprecation policy is less strict and just leaves it to the maintainers, but that may not be sensible given that conda is a larger project and it's good to default to clear and predictable schedules.

Reply:
You're absolutely right, I was not a fan of how this created a new 3 step schedule. Now that we've defined the concept of a deprecation release (instead of counting the number of releases), it makes a lot more sense as you suggested.